src/services/api/makeFMCError.ts

```diff
@@ -0,0 +1,71 @@
+import ky from 'ky-universal';
+import debug from 'debug';
+import { ApiError, NetworkError, TimeoutError } from './errors';
+import { FMCError, JSONValue, JSONObject } from './types';
+
+const log = debug('fmc:api:errorHandling');
+
+/**
+ * Translate errors reported by ky/fetch to a set of custom exceptions
+ * which we can later on use to make decisions on how to handle specific errors.
+ */
+export default async function makeFMCError(e: FMCError): Promise<FMCError> {
+  let translatedError: FMCError;
+  let errorMessage: string;
+  let statusCode: number;
+  switch (e.constructor) {
+    case ky.HTTPError: // a non 2xx error code was found
+      errorMessage = await parseErrorResponse(e.response);
+      statusCode = e.response.status;
+      translatedError = new ApiError(errorMessage, statusCode);
+      break;
+    case ky.TimeoutError:
+      translatedError = new TimeoutError(e.message);
+      break;
+    case TypeError:
+      translatedError = e;
+      break;
+    default:
+      // pain point: every unexpected Error instance (that is not a TypeError)
+      // will be handled as NetworkError
+      translatedError = new NetworkError(e);
+      break;
```

Comment:
This is the only thing I'm still not satisfied with. I want it to only throw a NetworkError if there actually was a network error.

Reply:
Actually it is not that hard.
We have to check for `e.message ===' Network request failed'` (I can't remember why I am using 'ConnectionError' in the mocked fetch rejection. I will add this within the next days.
I found a good article on the topic: https://medium.com/to-err-is-aaron/detect-network-failures-when-using-fetch-40a53d56e36